File: lessons/compiled_lang/java/src/e40.branch.java
import java.io.IOException;

class E40Branch {
    public static void main(String[] args) throws IOException {
        System.out.print(
            "Select an item from the menu.\n\n" +
            "  1 - Coffee\n" +
            "  2 - Espresso\n" +
            "  3 - Latte\n" +
            "  4 - Machiato\n" +
            "  5 - Capucino\n" +
            "  6 - Mocha\n" +
            "  7 - Tea\n\n" +
            "Make your selection: ");

        int selection = System.in.read() - '0';

        switch (selection) {
            case 1: System.out.println("You selected a Coffee"); break;
            case 2: System.out.println("You selected an Espresso"); break;
            case 3: System.out.println("You selected a Latte"); break;
            case 4: System.out.println("You selected a Machiato"); break;
            case 5: System.out.println("You selected a Capucino"); break;
            case 6: System.out.println("You selected a Mocha"); break;
            case 7: System.out.println("You selected a Tea"); break;
            default: System.out.println("You have not entered a valid selection");
        }
    }
}

File: lessons/compiled_lang/rust/src/e40.branch.rs
use std::io::{self, Read, Write};

fn main() {
    print!("Select an item from the menu.\n\n  1 - Coffee\n  2 - Espresso\n  3 - Latte\n  4 - Machiato\n  5 - Capucino\n  6 - Mocha\n  7 - Tea\n\nMake your selection: ");
    io::stdout().flush().unwrap();

    let mut buf = [0u8; 1];
    io::stdin().read_exact(&mut buf).unwrap();
    let selection = buf[0] as i32 - '0' as i32;

    match selection {
        1 => println!("You selected a Coffee"),
        2 => println!("You selected an Espresso"),
        3 => println!("You selected a Latte"),
        4 => println!("You selected a Machiato"),
        5 => println!("You selected a Capucino"),
        6 => println!("You selected a Mocha"),
        7 => println!("You selected a Tea"),
        _ => println!("You have not entered a valid selection"),
    }
}
